### ISSUE TITLE
module 'AWSS3' was not compiled for testing

### ISSUE THREAD

**Issue by chriswindings** [aws-amplify/amplify-swift#3877]
### Describe the bug

Running into error `module 'AWSS3' was not compiled for testing` when running `xcodebuild` in our CICD pipeline. 

### Steps To Reproduce

```swift
Build the app in CICD pipeline using fastlane to run xcodebuild
```

### Expected behavior

Expects to build the app archive, but fails to build, pointing to the usage of the `@testable` attribute as the cause of failure. 

### Amplify Framework Version

2.41.0

### Amplify Categories

Storage

### Dependency manager

Swift PM

### Swift version

5.8

### CLI version

12.11.1

### Xcode version

16.0

### Relevant log output

```shell
[15:06:56]: ▸ [AWSS3StoragePlugin] Copy AWSS3StoragePlugin.modulemap -> AWSS3StoragePlugin.modulemap
[15:06:56]: ▸ ::error file=/Users/runner/Library/Developer/Xcode/DerivedData/*****/SourcePackages/checkouts/amplify-swift/AmplifyPlugins/Storage/Sources/AWSS3StoragePlugin/Dependency/UploadPartInput+presignURL.swift,line=8,col=18::module 'AWSS3' was not compiled for testing
[15:06:56]: ▸ @testable import AWSS3
```

### Is this a regression?

Yes

### Regression additional context

2.40.0 works fine, issue introduced in 2.41.0

### Platforms

iOS

### OS Version

iOS 18 / macOS 15.0

### Device

iPhone 15 Pro

**Comment by xxza-cognizant** [comment]
I was going to raise the same issue, glad to see it has been fixed. Thanks Amplify team

**Comment by phantumcode** [comment]
Fixed in release [v2.41.1](https://github.com/aws-amplify/amplify-swift/releases/tag/2.41.1)
